pkg/image/reference/reference.go

```diff
@@ -0,0 +1,51 @@
+package reference
+
+import (
+	"strings"
+
+	"github.com/docker/distribution/reference"
+)
+
+// NamedDockerImageReference points to a Docker image.
+type NamedDockerImageReference struct {
```

Comment:
I split it to able use this parser in S2I as mentioned in https://github.com/openshift/origin/pull/11245#issuecomment-251974739. S2I can't use `ParseDockerImageReference` directly because `origin` imports it. To avoid differences in the parser behavior we need to use one parser everywhere. So I move it into separate module with minimal dependencies.